reviews/serializers.py

```diff
@@ -2,53 +2,78 @@
 from __future__ import unicode_literals
 
 from rest_framework import serializers as ser
+from rest_framework.fields import SkipField
+from transitions import MachineError
 
 from api.base.utils import absolute_reverse
+from api.base.utils import get_user_auth
 from api.base.serializers import JSONAPISerializer
 from api.base.serializers import RelationshipField
 from api.base.serializers import LinksField
 
+from reviews.workflow import Actions
+from reviews.workflow import States
+from reviews.workflow import Workflows
 from reviews.models import ReviewLog
-from reviews.models import ReviewProviderMixin
-from reviews.models import ReviewableMixin
+from reviews.permissions import has_reviews_permission
+
+
+# Pseudo-class to hide the creator field if it shouldn't be shown
+def IfNotAnonymous(field_cls):
+    class Field(field_cls):
+        def get_attribute(self, instance):
+            request = self.context.get('request')
+            view = self.context.get('view')
+            auth = get_user_auth(request)
+            if view and request and auth.logged_in:
+                comments_anonymous = instance.reviewable.provider.reviews_comments_anonymous
```

Comment:
`LogPermission` checks `reviews_comments_private` to see whether you can see the log at all, this checks `reviews_comments_anonymous` to see whether it should include `creator` in its response.